src/transaction/deploy_account.rs

```diff
@@ -204,34 +204,45 @@ impl DeployAccount {
         self.handle_nonce(state)?;
 
         let mut transactional_state = state.create_transactional()?;
-        let mut tx_exec_info = self.apply(
-            &mut transactional_state,
-            block_context,
-            #[cfg(feature = "cairo-native")]
-            program_cache.clone(),
-        )?;
-
-        let actual_fee = calculate_tx_fee(
-            &tx_exec_info.actual_resources,
-            block_context.starknet_os_config.gas_price,
-            block_context,
-        )?;
+        let mut tx_exec_info;
 
-        if let Some(revert_error) = tx_exec_info.revert_error.clone() {
-            // execution error
-            tx_exec_info = tx_exec_info.to_revert_error(&revert_error);
-        } else if actual_fee > self.max_fee {
-            // max_fee exceeded
-            tx_exec_info = tx_exec_info.to_revert_error(
-                format!(
-                    "Calculated fee ({}) exceeds max fee ({})",
-                    actual_fee, self.max_fee
-                )
-                .as_str(),
-            );
-        } else {
-            state
-                .apply_state_update(&StateDiff::from_cached_state(transactional_state.cache())?)?;
+        {
+            let _original_state = state.create_transactional()?;
```

Comment:
Transactional state is used for revertible transactions, we can just use the state